crates/router/src/core/payments/types.rs

```diff
@@ -212,8 +212,9 @@ impl ForeignTryFrom<(&SurchargeDetails, &PaymentAttempt)> for SurchargeDetailsRe
                 .tax_on_surcharge_amount
                 .get_amount_as_i64(),
         )?;
-        let display_final_amount = currency
-            .to_currency_base_unit_asf64(surcharge_details.final_amount.get_amount_as_i64())?;
+        let display_final_amount = currency.to_currency_base_unit_asf64(
```

Comment:
@hrithikesh026 Is this final_amount is total_amount including all amount related calculation or only order_amount + surcharge?

Reply:
Currently `display_final_amount` is not being used, removing it